### ISSUE TITLE
wasm-eh: wasm-ld error in relocation R_WASM_MEMORY_ADDR_SLEB cannot be used against symbol GCC_except_table0

### ISSUE THREAD

**Issue by sps-gold** [emscripten-core/emscripten#15171]
wasm-ld throws relocation error building with wasm-exceptions, for c++ catch without ellipse and using mainmodule flag

here is a small reproducer https://github.com/emscripten-core/emscripten/pull/15170 for the error we are seeing during building some of our tests

```
wasm-ld: error: lto.tmp: relocation R_WASM_MEMORY_ADDR_SLEB cannot be used against symbol GCC_except_table0; recompile with -fPIC

```
following is the link to [ci error](https://app.circleci.com/pipelines/github/emscripten-core/emscripten/15768/workflows/e4cb4679-fa55-48d8-b2eb-3dbf6989a715/jobs/459433?invite=true&invite=true#step-106-596)

**Comment by kripken** [comment]
@aheejin is this a known issue? Not sure it's specific to exceptions though, maybe it's more of a relocation linking issue...

**Comment by aheejin** [comment]
Yes, this is a known issue, and I'm working on the fix. Likely to be fixed within a few days.

**Comment by aheejin** [comment]
Sorry for the long delay, and the fix is in #15266 (and its companion patches). I'm not sure how you cherry-picked my patch, given that any of the patches (LLVM, Binaryen, and Emscripten) hasn't landed. Do you mean you downloaded each of the three patches and applied them and compiled LLVM and Binaryen yourself?

I think you at least need to wait until these patches land, and you can either use `emsdk install latest` do pick up the latest trunk, or wait until the next release. I'm not sure what you exactly mean by cherry-picking.

**Comment by sps-gold** [comment]
yes, that's right I took your changes and build the binaries myself. 

Thanks @aheejin for fixing the issue, this really helps us unblock ourself from further validation of the upcoming eh feature. 
I see I can now build and run the reproducer successfully, so now I can continue to build main app and uncover other issues if any.

```
~/Source/github/sps-gold/test$ ./runtime/d8_ver9.5/d8 --experimental-wasm-eh ./test_exceptions_std.js
what? std::exception
caught std::exception
```
